scripts/helpers/iwyu_check.py

```diff
@@ -0,0 +1,194 @@
+    # Ensures somewhat pretty logging of what is going on
+    log_fmt = '%(asctime)s %(levelname)-7s %(message)s'
+    if no_log_timestamps:
+        log_fmt = '%(levelname)-7s %(message)s'
+    coloredlogs.install(level=__LOG_LEVELS__[log_level], fmt=log_fmt)
+
+    # checking if a program IWYU exists
+    iwyu = find_program(('iwyu_tool', 'iwyu_tool.py'))
+
+    if iwyu is None:
+        logging.error("Can't find IWYU")
+        sys.exit(1)
+
+    # For iterating how many files had problems with includes
+    warning_in_files = 0
+
+    platform = ""
+    compile_commands_glob = glob.glob(compile_commands_glob)
+
+    if not compile_commands_glob:
+        logging.error("Can't find compile_commands.json file(s)")
+        sys.exit(1)
+
+    for compile_commands in compile_commands_glob:
+
+        compile_commands_path = os.path.dirname(compile_commands)
+        compile_commands_file = os.path.join(compile_commands_path, "compile_commands.json")
+        logging.info("Copy compile command file %s to %s" %
+                     (compile_commands, compile_commands_file))
+
+        with contextlib.suppress(shutil.SameFileError):
+            shutil.copyfile(compile_commands, compile_commands_file)
+
+        # Prase json file for find target name
+        with open(compile_commands, 'r') as json_data:
+            json_data = json.load(json_data)
+
+        for key in json_data:
+            find_re = re.search(r'^.*/src/platform*?\/(.*)/.*$', key['file'])
+            if find_re is not None:
+                platform = find_re.group(1)
+                break
+        if not platform:
+            logging.error("Can't find platform")
+            sys.exit(1)
+
+        if not mapping_file_dir:
+            mapping_file_dir = os.path.join(root_dir + "/platforms/" + platform)
+
+        # Specify config for platforms. Description: Some platform may needed hack clang compiler
+        if platform.lower() == "tizen":
+            if not clang_args:
+                clang_args = [
+                    "--target=arm-linux-gnueabi",
+                    "-I$TIZEN_SDK_TOOLCHAIN/arm-tizen-linux-gnueabi/include/c++/9.2.0",
+                    "-I$TIZEN_SDK_TOOLCHAIN/arm-tizen-linux-gnueabi/include/c++/9.2.0/arm-tizen-linux-gnueabi"
+                ]
+                clang_args = " ".join(clang_args)
+
+        # TODO: Add another platform for easy scanning
+        # Actually works scanning for platform: tizen, darwin, linux other not tested yet.
+
+        command_arr = [
+            iwyu,
+            "-p", compile_commands_path, scanning_destination,
+            "--", iwyu_args,
+            "-Xiwyu", "--mapping_file=" + mapping_file_dir + "/iwyu.imp",
+            clang_args
+        ]
+
+        logging.info("Start scanning include using IWYU for %s platform" % platform)
+        status = subprocess.Popen(" ".join(command_arr),
+                                  shell=True,
+                                  env=os.environ,
```

Comment:
Is `env` required here? By default it should be inherited from the parent process.